blockly-core/core/ui/block_space/block_space_editor.js

```diff
@@ -417,23 +418,34 @@ Blockly.BlockSpaceEditor.prototype.svgSize = function() {
  */
 Blockly.BlockSpaceEditor.prototype.svgResize = function() {
   var svg = this.svg_;
-  var style = window.getComputedStyle(svg);
-  var borderWidth = 0;
-  if (style) {
-    borderWidth = parseInt(style.borderLeftWidth, 10) +
-      parseInt(style.borderRightWidth, 10);
-  }
-  var div = svg.parentNode;
-  var width = div.offsetWidth - borderWidth;
-  var height = div.offsetHeight;
-  if (svg.cachedWidth_ != width) {
-    svg.setAttribute('width', width + 'px');
-    svg.cachedWidth_ = width;
-  }
-  if (svg.cachedHeight_ != height) {
-    svg.setAttribute('height', height + 'px');
-    svg.cachedHeight_ = height;
+  var svgStyle = window.getComputedStyle(svg);
+  var svgBorderWidth = 0;
+  if (svgStyle) {
+    svgBorderWidth = parseInt(svgStyle.borderLeftWidth, 10) +
+      parseInt(svgStyle.borderRightWidth, 10);
   }
+
+  // Subtract any pixels present above the svg element from the available height
+  // (only need to do this for mainBlockSpaceEditor's svg element, but fall back
+  // to this.svg_ during mainBlockSpaceEditor's creation)
+  var containerDiv = svg.parentNode;
+  var topmostSvgElement = Blockly.mainBlockSpaceEditor ? Blockly.mainBlockSpaceEditor.svg_ : svg;
```

Comment:
The modal editor would look ok, but would be too tall. If you are using page offset subtraction, you may not need to use the topmost SVG anymore.

Re: use of goog.style.getSize() - we need to make sure it is the inner size (clientWidth) and not the size including the border (offsetWidth) or else it will not quite fit in the container. I don't know what goog.style.getSize() returns.